examples/sentence-transformers-training/paraphrases/training_paraphrases.py

```diff
@@ -0,0 +1,149 @@
+    "stack-exchange": stack_exchange_train_dataset,
+}
+print(train_dataset_dict)
+
+# 1. Here we define our SentenceTransformer model. If not already a Sentence Transformer model, it will automatically
+# create one with "mean" pooling.
+model = SentenceTransformer(model_name)
+# If we want, we can limit the maximum sequence length for the model
+model.max_seq_length = max_seq_length
+logging.info(model)
+
+# 3. Define our training loss
+train_loss = MultipleNegativesRankingLoss(model)
+
+# 4. Define an evaluator for use during training. This is useful to keep track of alongside the evaluation loss.
+stsb_eval_dataset = load_dataset("sentence-transformers/stsb", split="validation")
+dev_evaluator = EmbeddingSimilarityEvaluator(
+    sentences1=stsb_eval_dataset["sentence1"],
+    sentences2=stsb_eval_dataset["sentence2"],
+    scores=stsb_eval_dataset["score"],
+    main_similarity=SimilarityFunction.COSINE,
+    name="sts-dev",
+)
+
+# 5. Define the training arguments
+args = SentenceTransformerGaudiTrainingArguments(
+    # Required parameter:
+    output_dir=output_dir,
+    # Optional training parameters:
+    num_train_epochs=num_epochs,
+    per_device_train_batch_size=batch_size,
+    per_device_eval_batch_size=batch_size,
+    warmup_ratio=0.1,
+    #fp16=True,  # Set to False if you get an error that your GPU can't run on FP16
+    #bf16=False,  # Set to True if you have a GPU that supports BF16
+    batch_sampler=BatchSamplers.NO_DUPLICATES,  # MultipleNegativesRankingLoss benefits from no duplicate samples in a batch
+    # We can use ROUND_ROBIN or PROPORTIONAL - to avoid focusing too much on one dataset, we will
+    # use round robin, which samples the same amount of batches from each dataset, until one dataset is empty
+    multi_dataset_batch_sampler=MultiDatasetBatchSamplers.ROUND_ROBIN,
+    # Optional tracking/debugging parameters:
+    evaluation_strategy="steps",
+    eval_steps=1000,
+    save_strategy="steps",
+    save_steps=1000,
+    save_total_limit=2,
+    logging_steps=100,
+    run_name="paraphrases-multi",  # Will be used in W&B if `wandb` is installed
+    use_habana = True,
+    gaudi_config_name = 'Habana/distilbert-base-uncased',
+    use_lazy_mode = True,
+    use_hpu_graphs = True,
+    use_hpu_graphs_for_inference = False,
+    use_hpu_graphs_for_training = True,
+
+)
+
+# 6. Create the trainer & start training
+trainer = SentenceTransformerGaudiTrainer(
+    model=model,
+    args=args,
+    train_dataset=train_dataset_dict,
+    loss=train_loss,
+    evaluator=dev_evaluator,
+)
+trainer.train()
+
+# 7. Evaluate the model performance on the STS Benchmark test dataset
+test_dataset = load_dataset("sentence-transformers/stsb", split="test")
+test_evaluator = EmbeddingSimilarityEvaluator(
+    sentences1=test_dataset["sentence1"],
+    sentences2=test_dataset["sentence2"],
+    scores=test_dataset["score"],
+    main_similarity=SimilarityFunction.COSINE,
+    name="sts-test",
+)
+#test_evaluator(model)
```

Comment:
Why is this commented?

Reply:
it could be uncomment and i forgot to uncomment. thanks

Reply:
uncommented done!